torchvision/models/_api.py

```diff
@@ -107,10 +110,26 @@ def get_weight(name: str) -> WeightsEnum:
     return weights_enum.from_str(value_name)
 
 
-def _get_enum_from_fn(fn: Callable) -> WeightsEnum:
+W = TypeVar("W", bound=Type[WeightsEnum])
+
+
+def get_model_weight(name: str) -> W:
```

Comment:
Maybe this could support a callable as well?

Reply:
I can make `_get_enum_from_fn` public or I can make this one accept both names and fns. I think having separate methods is cleaner. 

Unfortunately if I support both, we might have issues with the naming of the methods. I could perhaps name them like `get_model_weight_by_name()` and `get_model_weight_by_fn()`, but I'm not too happy with them. Any thoughts?

Reply:
Personally I'd go with a single function that accepts strings and callables (similarly to the `weights` parameters which accept strings and `Weight` objects)

Reply:
Done can you have one more look overall?